### PR TITLE
Groundwork for factions

### PR DESCRIPTION
**Pull request by IllustrisJack**
This is how far i feel comfortable going with implementing some kind of groundwork of faction functionality without the design being discussed fully. Also contains a fix for ship subtype of patrols (Zone Security) & a new Iterator function which can be used for numerical incremented tables. 

tracked-by: https://github.com/Limit-Theory-Redux/ltheory/issues/87

### CODE REVIEW

File: script/Core/Util/Guid.lua [hunk 1]
```diff
@@ -0,0 +1,11 @@
+function Guid()
```
Comment (IllustrisJack) [inline script/Core/Util/Guid.lua:1]: We should discuss if we wanna have a UUID Guid system or just use u64 (like the one in libphx right now)

File: script/Systems/Universe/UniverseEconomy.lua
Comment (Flatfingers) [inline script/Systems/Universe/UniverseEconomy.lua:1]: "Disposition" was a Josh feature that I used to support a very limited version of Faction. It enabled stations to shoot back at any ship that shot the station enough to make the station angry. I always assumed that Disposition would be replaced by a complete (even if basic) Faction implementation.

File: script/Enums/FactionEnums.lua
Comment (Flatfingers) [inline script/Enums/FactionEnums.lua:1]: Be aware that I've been developing space games for 30+ years now, and I have a _lot_ of FactionTypeNames defined over the years. :D

File: script/Systems/Universe/UniverseEconomy.lua
Comment (IllustrisJack) [reply script/Systems/Universe/UniverseEconomy.lua:1]: Yeah i´m also in favor in removing it, for now i was going to keep it (while transitioning) and drive a dual system until the full system is designed / factions are expanded

**Review by Flatfingers — APPROVED** [review]

File: script/Enums/EntityType.lua
[the remark this answers is not in the document]
Comment (IllustrisJack) [reply script/Enums/EntityType.lua:1]: Was also thinking about this for some time, depending on what we need we can write an entirely seperate system, but including it as an actual "physical" entity in the world would also bring some nice things with it. Noted for discussion once we touch this again :)

File: script/Enums/FactionEnums.lua
Comment (IllustrisJack) [reply script/Enums/FactionEnums.lua:1]: hahah looking forward to the 3000+lines file 🤣 - the more the better tho, atleast as long we can actually define some behavioural differences between them at some point

**Pull request merged**